src/java.desktop/windows/native/libawt/windows/Devices.cpp

```diff
@@ -92,53 +92,64 @@ int g_nMonitorCounter;
 int g_nMonitorLimit;
 HMONITOR* g_hmpMonitors;
```

Comment:
These global variables — `g_nMonitorCounter`, `g_nMonitorLimit`, `g_hmpMonitors` — could be `static` as well.

In fact, they can be encapsulated in the relevant functions which need them by using the last parameter of [`EnumDisplayMonitors`](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-enumdisplaymonitors). For `CountMonitors`, using `&nMonitorCounter` will be enough; for `CollectMonitors`, the three variables need to be encapsulated in a local structure the pointer to which is passed.

However, it's likely not worth the effort… If it works, don't touch it.

Reply:
Good suggestion, I created local variables and pass the data on to the callback functions.